From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Thu, 1 Jun 2023 19:30:16 +0200
Subject: [PATCH] Configuration - Get Fiddle configuration with Bukkit API

Allows getting the Fiddle configuration using the Bukkit API.
This is done by having `CraftServer` implement `Server.getFiddleConfig()`, which is defined in the "Configuration - Get Fiddle configuration with Bukkit API" patch in the `fiddle-api` module.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

This patch is based on the following patch:
"Gale configuration"
By: Martijn Muijsers <martijnmuijsers@live.nl>
As part of: Gale (https://github.com/GaleMC/Gale)
Licensed under: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)

* Gale description *

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Gale - https://galemc.org

This patch is based on the following patch:
"Paper config files"
By: Jake Potrebic <jake.m.potrebic@gmail.com>
As part of: Paper (https://github.com/PaperMC/Paper)
Licensed under: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)

diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index fd7d812c1e7d8928645259422c2bad0bb3d748c2..e931a3a47226ff7b8b42eb35d7bb0bf99dedf301 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -3012,6 +3012,14 @@ public final class CraftServer implements Server {
             return CraftServer.this.console.paperConfigurations.createLegacyObject(CraftServer.this.console);
         }
 
+        // Fiddle start - Fiddle configuration - API
+        @Override
+        public YamlConfiguration getFiddleConfig()
+        {
+            return org.fiddlemc.fiddle.configuration.FiddleConfigurations.get(null).createLegacyObject(CraftServer.this.console);
+        }
+        // Fiddle end - Fiddle configuration - API
+
         @Override
         public void restart() {
             org.spigotmc.RestartCommand.restart();
